benchmarks/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -387,8 +388,10 @@ int main_(std::vector<std::string> const &args, const MPI_Comm comm)
       for (int i = 0; i < n_queries; ++i)
         random_queries_host(i) = {
             {a * ((offset_x + random()) / 3 + max_offset / 3),
-             a * ((offset_y + random()) / 3 + max_offset / 3),
-             a * ((offset_z + random()) / 3 + max_offset / 3)}};
+             a * ((offset_y + random()) / 3 + max_offset / 3) *
+                 (partition_dim > 1),
```

Comment:
Hmm... I would prefer not using implicit conversion between `bool` and built-in types and prefer ternary operators.

In the end, it doesn't matter if we always fill the second and third component, though, does it?

Reply:
It matters. We have to fill them with zeros. Otherwise, the distances (and thus density) change because we have an additional vertical distance to consider.

Reply:
Ah, right. I was thinking for a moment that we only copy the needed components for some reason.